deadlines-api/src/main/java/com/palantir/deadlines/api/DeadlineExpiredException.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.deadlines.api;
+
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.Safe;
+import com.palantir.logsafe.SafeLoggable;
+import java.util.List;
+
+/**
+ * Indicates that a deadline has expired.
+ */
+public abstract sealed class DeadlineExpiredException extends RuntimeException implements SafeLoggable {
```

Comment:
I'm not 100% sure the exceptions make sense in a separate jar from the other parts of the deadliens library, it's all the deadlines API. Maybe if we put the exceptions in a package dedicated to exceptions? But I suspect the ability to check if there's remaining deadline time and the exceptions should live in the same package. Maybe we throw it all in `deadlines` and drop `deadlines-api`?

Reply:
ya that's fair, i'm on board with just removing deadlines-api entirely as part of this change.